### PR TITLE
Customer Controller - checkout

### PR DESCRIPTION
**Pull request by Hassan-AbuGareeb**
implemented and tested the checkout functionality

### CODE REVIEW

File: server/server/controllers/customer.js [hunk 1]
```diff
@@ -186,12 +186,52 @@ async function deleteCart(req, res) {
...
+async function checkout(req, res) {
+  const customerId = req.user.userId;
+  try {
+    //get the customer and fill the basket with the items data
+    const customer = await customerModel
+      .findById(customerId)
+      .populate("basket.items.itemId");
+    const basketItems = customer.basket.items;
+    if (basketItems.length < 1) {
+      return res.status(301).json({ message: "basket is empty!" });
```
Comment (Shwarzar) [inline server/server/controllers/customer.js:198]: This should be handled in frontend phase where we redirect the user to homepage instead

File: server/server/controllers/customer.js [hunk 2]
```diff
@@ -186,12 +186,52 @@ async function deleteCart(req, res) {
...
+    }
+    //create the order object and fill it with data from the basket
+    const order = {};
```
Comment (Shwarzar) [inline server/server/controllers/customer.js:201]: Required restaurantId from the items in the basket that will be ordered

**Review by Shwarzar — CHANGES_REQUESTED** [review]
Changes requested

File: server/server/controllers/customer.js (hunk 2 above)
Comment (Hassan-AbuGareeb) [reply server/server/controllers/customer.js:202]: added and pushed

**Review by luj00 — APPROVED** [review]

File: server/server/controllers/customer.js [hunk 3]
```diff
@@ -218,6 +220,7 @@ async function checkout(req, res) {
     basket.quantity = 0;
     customer.basket = basket;
     await customer.save();
+    console.log(customer.orders);
```
Comment (Shwarzar) [inline server/server/controllers/customer.js:223]: You can delete that since the function is working

File: server/server/models/customer.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const orderSchema = new mongoose.Schema({
   restaurantId: {
     type: mongoose.Schema.Types.ObjectId,
     ref: "Restaurant",
-    required: true,
+    // required: true,
```
Comment (Shwarzar) [inline server/server/models/customer.js:7]: Uncomment this line

**Review by Shwarzar — APPROVED** [review]
Approving changes

**Review by Shwarzar — APPROVED** [review]
Approving changes

**Pull request merged**